client/question/services/EventHandlerServiceSpec.js

```diff
@@ -232,5 +189,26 @@ describe('EventHandlerService', function() {
         EventHandlerService.sendCurrentEventBatch();
         expect(EventHandlerService._getCurrentEventBatchLength()).toEqual(0);
       });
+
+    it('does not send the batch if any of sId / qId / qVer are null',
```

Comment:
Where in this test do we check that the batch isn't sent? I'd expect it to spy on something and verify it never gets called.

Also, at the end, perhaps init() it correctly and check that the batch actually does get sent in that case?

Reply:
If the batch were sent it would trigger an httprequest that the test would be responsible for catching (see other send batch tests). That doesn't occur, and we see that the batch keeps getting longer as additional events aren't being sent.

Technically the other tests check that (as it's init'ed correctly in the beforeEach). Happy to add another check if you would like.

Reply:
Nope, that's fine I think. Thanks for the explanation!